lib/ecto/repo/preloader.ex

```diff
@@ -272,6 +272,14 @@ defmodule Ecto.Repo.Preloader do
     We expected a tuple but we got: #{inspect(entry)}
     """
 
+  defp preload_order(assoc, query, related_field) do
+    custom_order_by = assoc.preload_order |> Enum.map(fn {direction, field} ->
```

Comment:
It is safe to assume that if cardinality is `many`, it will either be a `Ecto.Association.ManyToMany` or `Ecto.Association.Has`?